desktop_version/src/UtilityClass.cpp

```diff
@@ -71,21 +73,61 @@ int ss_toi(const std::string& str)
 	return retval;
 }
 
-std::vector<std::string> split( const std::string &s, char delim, std::vector<std::string> &elems )
-{
-	std::stringstream ss(s);
-	std::string item;
-	while(std::getline(ss, item, delim))
+bool next_split(
+	size_t* start,
+	size_t* len,
+	const char* str,
+	const char delim
+) {
+	size_t idx = 0;
+	*len = 0;
+
+	if (str[idx] == '\0')
 	{
-		elems.push_back(item);
+		return false;
+	}
+
+	while (true)
+	{
+		if (str[idx] == delim)
+		{
+			++*start;
+			return true;
+		}
+		else if (str[idx] == '\0')
+		{
+			return true;
+		}
+
+		++idx;
```

Comment:
The ++ use here and slightly above threw me off for a second... I know it's against style everywhere else but += 1 might be easier to read for the pointer derefs in particular.

Reply:
Done. Rebased and force-pushed.